password_security/tests/test_res_users.py

```diff
@@ -67,9 +72,34 @@ def test_check_password_returns_true_for_valid_password(self):
         )
 
     def test_check_password_raises_error_for_invalid_password(self):
+        self.main_comp.password_length = 12
         with self.assertRaises(UserError):
             self.rec_id._check_password("password")
 
+    def test_check_password_lower(self):
+        self.rec_id._check_password("AB3!")
+        self.main_comp.password_lower = 1
+        with self.assertRaises(UserError):
+            self.rec_id._check_password("AB3!")
```

Comment:
I see all the tests are calling `with self.assertRaises(UserError)` without any distinction. So actually there's no check that the correct error message is shown.

Making use of `self.assertRaisesRegex()` would improve the tests, ensuring that the expected error is raised.

For example here it would be something like:

```python
        with self.assertRaisesRegex(UserError, ".*Lowercase letter (at least %s characters).*" % str(company_id.password_lower)):
            self.rec_id._check_password("AB3!")
```

What do you think?